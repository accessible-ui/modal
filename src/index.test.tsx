/* jest */
import React from 'react'
import {renderHook} from '@testing-library/react-hooks'
import {render, fireEvent} from '@testing-library/react'
import {
  Modal,
  Trigger,
  Target,
  Close,
  useControls,
  useIsOpen,
  useModal,
} from './index'

const click_ = fireEvent.click
fireEvent.click = (...args) => {
  fireEvent.mouseDown(...args)
  return click_(...args)
}

describe('<Modal>', () => {
  it('should have a custom id', () => {
    const result = render(
      <Modal id="foobar">
        <Target>
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Modal>
    )

    expect(result.asFragment()).toMatchSnapshot()
  })

  it('should provide context to function child', () => {
    let cxt

    render(
      <Modal>
        {context => {
          cxt = context
          return <div />
        }}
      </Modal>
    )

    expect(cxt).toMatchSnapshot()
  })
})

describe('<Target>', () => {
  it('should open and close on Trigger click', () => {
    const result = render(
      <Modal>
        <Target>
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Modal>
    )

    expect(result.asFragment()).toMatchSnapshot('closed initially')
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('open')
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('closed')
  })

  it('should close on escape key', () => {
    const result = render(
      <Modal>
        <Target>
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Modal>
    )

    expect(result.asFragment()).toMatchSnapshot('closed initially')
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('open')
    fireEvent.keyDown(result.getByText('Hello world'), {
      key: 'Escape',
      code: 27,
    })
    expect(result.asFragment()).toMatchSnapshot('closed')
  })

  it(`shouldn't close on escape key if prop is false`, () => {
    const result = render(
      <Modal>
        <Target closeOnEscape={false}>
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Modal>
    )

    expect(result.asFragment()).toMatchSnapshot('closed initially')
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('open')
    fireEvent.keyDown(result.getByText('Hello world'), {
      key: 'Escape',
      code: 27,
    })
    expect(result.asFragment()).toMatchSnapshot('still open')
  })

  it(`should assign to custom styles when opened or closed`, () => {
    const result = render(
      <Modal>
        <Target>
          <div style={{fontSize: '2rem'}}>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Modal>
    )

    expect(result.asFragment()).toMatchSnapshot()
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('open')
  })

  it(`should apply custom classname when opened or closed`, () => {
    const result = render(
      <Modal>
        <Target>
          <div className="custom">Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Modal>
    )

    expect(result.asFragment()).toMatchSnapshot()
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('open')
  })

  it(`should apply user defined openClass and closedClass`, () => {
    const result = render(
      <Modal>
        <Target closedClass="closed" openClass="open">
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Modal>
    )

    expect(result.asFragment()).toMatchSnapshot()
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('open')
  })

  it(`should apply user defined openStyle and closedStyle`, () => {
    const result = render(
      <Modal>
        <Target closedStyle={{display: 'none'}} openStyle={{display: 'block'}}>
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Modal>
    )

    expect(result.asFragment()).toMatchSnapshot()
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('open')
  })

  it(`should be initially open when defined as such`, () => {
    const result = render(
      <Modal defaultOpen>
        <Target>
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Modal>
    )

    expect(result.asFragment()).toMatchSnapshot('initially open')
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('closed')
  })

  it(`should act like a controlled component when 'open' prop is specified`, () => {
    const result = render(
      <Modal open>
        <Target>
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Modal>
    )

    expect(result.asFragment()).toMatchSnapshot('initially open')
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('still open')

    result.rerender(
      <Modal open={false}>
        <Target>
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Modal>
    )

    expect(result.asFragment()).toMatchSnapshot('closed')
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('still closed')
  })

  it('should render into a portal by default ID', () => {
    const portalRoot = document.createElement('div')
    portalRoot.setAttribute('id', 'portals')
    document.body.appendChild(portalRoot)

    const result = render(
      <Modal open>
        <Target portal>
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Modal>
    )

    fireEvent.click(result.getByText('open me'))
    expect(result.baseElement).toMatchSnapshot()
    document.body.removeChild(portalRoot)
  })

  it('should render into a portal by custom selector object', () => {
    const portalRoot = document.createElement('div')
    portalRoot.setAttribute('class', 'portals')
    document.body.appendChild(portalRoot)

    const result = render(
      <Modal open>
        <Target portal={{container: '.portals'}}>
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Modal>
    )

    fireEvent.click(result.getByText('open me'))
    expect(result.baseElement).toMatchSnapshot()
    document.body.removeChild(portalRoot)
  })

  it('should render into a portal by custom selector', () => {
    const portalRoot = document.createElement('div')
    portalRoot.setAttribute('class', 'portals')
    document.body.appendChild(portalRoot)

    const result = render(
      <Modal open>
        <Target portal=".portals">
          <div>Hello world</div>
        </Target>

        <Trigger>
          <button>open me</button>
        </Trigger>
      </Modal>
    )

    fireEvent.click(result.getByText('open me'))
    expect(result.baseElement).toMatchSnapshot()
    document.body.removeChild(portalRoot)
  })

  it('should override role', () => {
    const result = render(
      <Modal>
        <Target>
          <div role='other-role'>Hello world</div>
        </Target>
      </Modal>
    )

    expect(result.asFragment()).toMatchSnapshot()
  })

  it('should override aria-modal', () => {
    const result = render(
      <Modal>
        <Trigger>
          <div aria-modal='true'>Hello world</div>
        </Trigger>
      </Modal>
    )

    expect(result.asFragment()).toMatchSnapshot()
  })
})

describe('<Trigger>', () => {
  it('should have openClass and closedClass', () => {
    const result = render(
      <Modal>
        <Target>
          <div>Hello world</div>
        </Target>

        <Trigger closedClass="closed" openClass="open">
          <button>open me</button>
        </Trigger>
      </Modal>
    )

    expect(result.asFragment()).toMatchSnapshot()
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('open')
  })

  it('should have openStyle and closedStyle', () => {
    const result = render(
      <Modal>
        <Target>
          <div>Hello world</div>
        </Target>

        <Trigger closedStyle={{display: 'none'}} openStyle={{display: 'block'}}>
          <button>open me</button>
        </Trigger>
      </Modal>
    )

    expect(result.asFragment()).toMatchSnapshot()
    fireEvent.click(result.getByText('open me'))
    expect(result.asFragment()).toMatchSnapshot('open')
  })

  it('should override aria-haspopup', () => {
    const result = render(
      <Modal>
        <Trigger>
          <div aria-haspopup='grid'>Hello world</div>
        </Trigger>
      </Modal>
    )

    expect(result.asFragment()).toMatchSnapshot()
  })
})

describe('<Close>', () => {
  it('should close the modal', () => {
    const result = render(
      <Modal defaultOpen={true}>
        <Target>
          <div>
            <Close>
              <button data-testid="close">Close me</button>
            </Close>
            Hello world
          </div>
        </Target>

        <Trigger closedClass="closed" openClass="open">
          <button>open me</button>
        </Trigger>
      </Modal>
    )

    expect(result.asFragment()).toMatchSnapshot()
    fireEvent.click(result.getByTestId('close'))
    expect(result.asFragment()).toMatchSnapshot('closed')
  })
})

describe('useControls()', () => {
  it('should have toggle, open, close keys', () => {
    const {result} = renderHook(() => useControls(), {wrapper: Modal})
    expect(Object.keys(result.current)).toStrictEqual([
      'open',
      'close',
      'toggle',
    ])
  })
})

describe('useIsOpen()', () => {
  it('should return boolean', () => {
    const {result} = renderHook(() => useIsOpen(), {wrapper: Modal})
    expect(typeof result.current).toBe('boolean')
  })
})

describe('useModal()', () => {
  it('should return context', () => {
    const {result} = renderHook(() => useModal(), {wrapper: Modal})
    expect(result.current).toMatchSnapshot()
  })
})
