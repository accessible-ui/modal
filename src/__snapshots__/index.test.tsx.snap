// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Close> should close the modal 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    aria-modal="false"
    id="ðŸ…°16"
    role="dialog"
    style="visibility: visible; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    <button
      aria-controls="ðŸ…°16"
      aria-expanded="true"
      aria-label="Close"
      data-testid="close"
    >
      Close me
    </button>
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°16"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="open"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Close> should close the modal: closed 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    aria-modal="false"
    id="ðŸ…°16"
    role="dialog"
    style="visibility: hidden; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    <button
      aria-controls="ðŸ…°16"
      aria-expanded="false"
      aria-label="Close"
      data-testid="close"
    >
      Close me
    </button>
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°16"
    aria-expanded="false"
    aria-haspopup="dialog"
    class="closed"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Modal> should have a custom id 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    aria-modal="false"
    id="foobar"
    role="dialog"
    style="visibility: hidden; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="foobar"
    aria-expanded="false"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Modal> should provide context to function child 1`] = `
Object {
  "close": [Function],
  "id": "ðŸ…°1",
  "isOpen": false,
  "open": [Function],
  "toggle": [Function],
}
`;

exports[`<Target> should act like a controlled component when 'open' prop is specified: closed 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    aria-modal="false"
    id="ðŸ…°10"
    role="dialog"
    style="visibility: hidden; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°10"
    aria-expanded="false"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should act like a controlled component when 'open' prop is specified: initially open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    aria-modal="false"
    id="ðŸ…°10"
    role="dialog"
    style="visibility: visible; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°10"
    aria-expanded="true"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should act like a controlled component when 'open' prop is specified: still closed 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    aria-modal="false"
    id="ðŸ…°10"
    role="dialog"
    style="visibility: hidden; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°10"
    aria-expanded="false"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should act like a controlled component when 'open' prop is specified: still open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    aria-modal="false"
    id="ðŸ…°10"
    role="dialog"
    style="visibility: visible; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°10"
    aria-expanded="true"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should apply custom classname when opened or closed 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    aria-modal="false"
    class="custom"
    id="ðŸ…°6"
    role="dialog"
    style="visibility: hidden; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°6"
    aria-expanded="false"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should apply custom classname when opened or closed: open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    aria-modal="false"
    class="custom"
    id="ðŸ…°6"
    role="dialog"
    style="visibility: visible; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°6"
    aria-expanded="true"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should apply user defined openClass and closedClass 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    aria-modal="false"
    class="closed"
    id="ðŸ…°7"
    role="dialog"
    style="visibility: hidden; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°7"
    aria-expanded="false"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should apply user defined openClass and closedClass: open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    aria-modal="false"
    class="open"
    id="ðŸ…°7"
    role="dialog"
    style="visibility: visible; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°7"
    aria-expanded="true"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should apply user defined openStyle and closedStyle 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    aria-modal="false"
    id="ðŸ…°8"
    role="dialog"
    style="visibility: hidden; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0); display: none;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°8"
    aria-expanded="false"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should apply user defined openStyle and closedStyle: open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    aria-modal="false"
    id="ðŸ…°8"
    role="dialog"
    style="visibility: visible; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0); display: block;"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°8"
    aria-expanded="true"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should assign to custom styles when opened or closed 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    aria-modal="false"
    id="ðŸ…°5"
    role="dialog"
    style="visibility: hidden; font-size: 2rem; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°5"
    aria-expanded="false"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should assign to custom styles when opened or closed: open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    aria-modal="false"
    id="ðŸ…°5"
    role="dialog"
    style="visibility: visible; font-size: 2rem; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°5"
    aria-expanded="true"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should be initially open when defined as such: closed 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    aria-modal="false"
    id="ðŸ…°9"
    role="dialog"
    style="visibility: hidden; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°9"
    aria-expanded="false"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should be initially open when defined as such: initially open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    aria-modal="false"
    id="ðŸ…°9"
    role="dialog"
    style="visibility: visible; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°9"
    aria-expanded="true"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should close on escape key: closed 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    aria-modal="false"
    id="ðŸ…°3"
    role="dialog"
    style="visibility: visible; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°3"
    aria-expanded="true"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should close on escape key: closed initially 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    aria-modal="false"
    id="ðŸ…°3"
    role="dialog"
    style="visibility: hidden; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°3"
    aria-expanded="false"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should close on escape key: open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    aria-modal="false"
    id="ðŸ…°3"
    role="dialog"
    style="visibility: visible; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°3"
    aria-expanded="true"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should open and close on Trigger click: closed 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    aria-modal="false"
    id="ðŸ…°2"
    role="dialog"
    style="visibility: hidden; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°2"
    aria-expanded="false"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should open and close on Trigger click: closed initially 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    aria-modal="false"
    id="ðŸ…°2"
    role="dialog"
    style="visibility: hidden; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°2"
    aria-expanded="false"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should open and close on Trigger click: open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    aria-modal="false"
    id="ðŸ…°2"
    role="dialog"
    style="visibility: visible; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°2"
    aria-expanded="true"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> should render into a portal by custom selector 1`] = `
<body>
  <div
    class="portals"
  >
    <div
      aria-hidden="false"
      aria-modal="false"
      id="ðŸ…°13"
      role="dialog"
      style="visibility: visible; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
    >
      Hello world
    </div>
  </div>
  <div>
    <button
      aria-controls="ðŸ…°13"
      aria-expanded="true"
      aria-haspopup="dialog"
    >
      open me
    </button>
  </div>
</body>
`;

exports[`<Target> should render into a portal by custom selector object 1`] = `
<body>
  <div
    class="portals"
  >
    <div
      aria-hidden="false"
      aria-modal="false"
      id="ðŸ…°12"
      role="dialog"
      style="visibility: visible; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
    >
      Hello world
    </div>
  </div>
  <div>
    <button
      aria-controls="ðŸ…°12"
      aria-expanded="true"
      aria-haspopup="dialog"
    >
      open me
    </button>
  </div>
</body>
`;

exports[`<Target> should render into a portal by default ID 1`] = `
<body>
  <div
    id="portals"
  >
    <div
      aria-hidden="false"
      aria-modal="false"
      id="ðŸ…°11"
      role="dialog"
      style="visibility: visible; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
    >
      Hello world
    </div>
  </div>
  <div>
    <button
      aria-controls="ðŸ…°11"
      aria-expanded="true"
      aria-haspopup="dialog"
    >
      open me
    </button>
  </div>
</body>
`;

exports[`<Target> shouldn't close on escape key if prop is false: closed initially 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    aria-modal="false"
    id="ðŸ…°4"
    role="dialog"
    style="visibility: hidden; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°4"
    aria-expanded="false"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> shouldn't close on escape key if prop is false: open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    aria-modal="false"
    id="ðŸ…°4"
    role="dialog"
    style="visibility: visible; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°4"
    aria-expanded="true"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Target> shouldn't close on escape key if prop is false: still open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    aria-modal="false"
    id="ðŸ…°4"
    role="dialog"
    style="visibility: visible; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°4"
    aria-expanded="true"
    aria-haspopup="dialog"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Trigger> should have openClass and closedClass 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    aria-modal="false"
    id="ðŸ…°14"
    role="dialog"
    style="visibility: hidden; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°14"
    aria-expanded="false"
    aria-haspopup="dialog"
    class="closed"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Trigger> should have openClass and closedClass: open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    aria-modal="false"
    id="ðŸ…°14"
    role="dialog"
    style="visibility: visible; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°14"
    aria-expanded="true"
    aria-haspopup="dialog"
    class="open"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Trigger> should have openStyle and closedStyle 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    aria-modal="false"
    id="ðŸ…°15"
    role="dialog"
    style="visibility: hidden; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°15"
    aria-expanded="false"
    aria-haspopup="dialog"
    style="display: none;"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`<Trigger> should have openStyle and closedStyle: open 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    aria-modal="false"
    id="ðŸ…°15"
    role="dialog"
    style="visibility: visible; position: fixed; margin: auto; left: 50%; top: 50%; transform: translate3d(-50%, -50%, 0);"
  >
    Hello world
  </div>
  <button
    aria-controls="ðŸ…°15"
    aria-expanded="true"
    aria-haspopup="dialog"
    style="display: block;"
  >
    open me
  </button>
</DocumentFragment>
`;

exports[`useModal() should return context 1`] = `
Object {
  "close": [Function],
  "id": "ðŸ…°19",
  "isOpen": false,
  "open": [Function],
  "toggle": [Function],
}
`;
